components/style/rule_tree/mod.rs

```diff
@@ -580,13 +586,13 @@ impl StrongRuleNode {
                 let existing =
                     next_sibling_ptr.compare_and_swap(ptr::null_mut(),
                                                       new_ptr,
-                                                      Ordering::SeqCst);
+                                                      Ordering::Acquire);
```

Comment:
Why? Upon reflection, I think even relaxed may be ok given we're looping. Aren't you mixing reads of the fields on the nodes vs reads on the atomic pointers? How's the field accesses on `existing` relevant to the loading order of `next_sibling_ptr`? (It's late and I may need re-paging all the memory orderings back in, but...)